conda/base/context.py

```diff
@@ -661,35 +659,44 @@ def signing_metadata_url_base(self):
 
     @property
     def conda_exe_vars_dict(self):
-        '''
-        An OrderedDict so the vars can refer to each other if necessary.
+        """
+        A dict so the vars can refer to each other if necessary.
         None means unset it.
-        '''
 
+        These are used by the initialization logic
+        (conda.core.initialize) to fill in the templates.
+        PR#11970 changed default use of CONDA_EXE (the entry point) to
+        'CONDA_PYTHON_EXE -I -m conda $@' to avoid shebang issues.
+
+        Note some Activator classes may extend this on their own
+        with _CONDA_EXE and _CONDA_ROOT (legacy variables). Adding
+        those here causes legacy test breakage because they inject
+        their own _CONDA_ROOT variables.
+        """
+        bin_dir = "Scripts" if on_win else "bin"
+        exe = "conda.exe" if on_win else "conda"
+        env_vars = {
+            "CONDA_EXE": os.path.join(sys.prefix, bin_dir, exe),
+            # -I enables 'isolated' mode: no local or user paths are searched
+            # for modules (only in $PREFIX/lib/pythonX.Y/*); it also ignores
+            # all PYTHON* env vars
+            "_CE_I": "-I",
+            "_CE_M": "-m",
+            "_CE_CONDA": "conda",
```

Comment:
I'd argue we don't rely on this environment variables anymore. They were needed before, when `__conda_exe()` and friends were defined like this:

```bash
$CONDA_EXE $_CE_M $_CE_CONDA $@
```

In normal circumstances, these variables were defined as:

* `CONDA_EXE`: path to `$BASE_PREFIX/bin/conda` (Unix) or `$BASE_PREFIX/Scripts/conda.exe` (Windows) -- in other words, the Python entry points.
* `_CE_M` and `_CE_CONDA`: empty string

In `dev` mode, they would change to
* `CONDA_EXE`: path to `python` executable in the base environment
* `_CE_M`: `-m`
* `_CE_CONDA`: `conda`

With this hack, we could do `conda.exe install ...` in normal mode and `python -m conda install ...`  in dev mode. However, since we are now ALWAYS using `python -m conda`, it makes no sense to provide this option. If anything, it can create issues because the user can accidentally delete or redefine `_CE_M` and `_CE_CONDA` and break the integration.

What I propose is to leave it like:

```
$CONDA_PYTHON_EXE $_CE_I -m conda $@
```

We still need to configure whether to use `-I` or not; we do use it for normal mode, but we omit it for dev mode so PYTHONPATH can be read. Right now, all the code in this PR assumes `_CE_M` and `_CE_CONDA` are always defined, but I'd say we just hardcode it because these do not need to be configurable.